client/reader_slice_of_slices_iterator.go

```diff
@@ -98,8 +118,20 @@ func (it *readerSliceOfSlicesIterator) Close() {
 		return
 	}
 	it.closed = true
-	if it.pool != nil {
-		it.pool.Put(it)
+	// Release any refs to segments
+	it.segments = nil
+	// Release any refs to segment byte slices
+	for i := range it.segmentReaders {
+		seg := it.segmentReaders[i].Segment()
+		if seg.Head != nil {
+			seg.Head.Reset(nil)
```

Comment:
So calling `Finalize(...)` on the `checked.Bytes` means it will return itself to a pool if its pooled.  Here we aren't using pooled `checked.Bytes` hence the use of reset and wanting to avoid any `Finalize(...)` calls.